SweetTweetApp/src/main/java/servletsForPosts/ServletPosts.java

```diff
@@ -42,35 +70,43 @@ protected void doDelete(HttpServletRequest request, HttpServletResponse response
     }
 
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        Post post = Post.fromJson(request.getParameter("info"));
-        PostCollection postCollection = new PostCollection();
+        Post post = Post.fromJson(IOUtils.toString(request.getReader()));
 
-        if (post == null) {
-            sendError(400, "Invalidate post", response);
+        PostCollection postCollection = new PostCollection();
+        post.setId(PostCollection.getAvailableId());
+        if (!checkAvailabilityUser(request, post.getAuthor())) {
+            sendError(401, "Post can\'t be added", response);
             return;
         }
 
-        if (PostCollection.hasPost(post.getId())) {
+        if (!PostCollection.hasPost(post.getId())) {
             if (PostCollection.addPost(post)) {
                 sendMessage(200, "Post was added", response);
             } else {
+                System.out.println(">>> INVALIDATE POST");
                 sendError(400, "Invalidate post", response);
             }
         } else {
+            System.out.println(">>> post is exist");
```

Comment:
Generally, you've made enough advance in your programming skills to start using logger. No one does `system.out.println` in production. It's good to master `system.util.logging`, or `slf4j`, or `logback` approach.
A logger will anyway write to console apart from where it is ought to write, so you won't lose visual feedback